File: Basic Programs -1/largest difference between two adjacent digits/largest difference.java
import java.util.Scanner;

public class LargestDifference {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String n = sc.next();
        int max_diff = 0;
        for(int i = 1; i < n.length(); i++) {
            int diff = Math.abs(n.charAt(i) - n.charAt(i-1));
            if(diff > max_diff) max_diff = diff;
        }
        System.out.println(max_diff);
    }
}

File: Basic Programs -1/largest difference between two adjacent digits/largest difference.c
#include<stdio.h>
#include<stdlib.h>

int main() {
    char n[10];
    scanf("%s", n);
    int max_diff = 0;
    for(int i = 1; n[i] != '\0'; i++) {
        int diff = abs(n[i] - n[i-1]);
        if(diff > max_diff) max_diff = diff;
    }
    printf("%d", max_diff);
    return 0;
}
